src/arch/ia32e/address.rs

```diff
@@ -0,0 +1,84 @@
+pub struct VirtualAddress(u64);
+
+impl VirtualAddress {
+    /// Try to convert u64 into an ia32e virtual address.
+    ///
+    /// # Safety
+    ///
+    /// This function ensures the virtual address is correct.
+    pub fn try_new(addr: u64) -> Option<Self>
+    {
+        if addr < (1 << 48) {
+            Some(VirtualAddress(addr))
+        } else {
+            None
+        }
+    }
+
+    /// Convert u64 into an ia32e virtual address.
+    ///
+    /// # Safety
+    ///
+    /// This function panics if the virtual address is incorrect.
+    pub fn new(addr: u64) -> Self
+    {
+        Self::try_new(addr).expect("Virtual address cannot be longer than 48 bits.")
+    }
+}
+
+pub struct PhysicalAddress(u64);
+
+impl PhysicalAddress {
+    /// Try to convert u64 into an ia32e physical address.
+    ///
+    /// # Safety
+    ///
+    /// This function ensures the physical address is correct.
+    pub fn try_new(addr: u64) -> Option<Self>
+    {
+        if addr < (1 << 52) {
+            Some(PhysicalAddress(addr))
+        } else {
+            None
+        }
+    }
+
+    /// Convert u64 into an ia32e physical address.
+    ///
+    /// # Safety
+    ///
+    /// This function panics if the physical address is incorrect.
+    pub fn new(addr: u64) -> Self
+    {
+        Self::try_new(addr).expect("Physical address cannot be longer than 52 bits.")
+    }
+}
+
```

Comment:
You might want to implement `Debug` trait as well as different formatting options for those types. It would be useful to be able to display the hexadecimal representation of an address.